### PR TITLE
Refactor `site` to make it easier to introduce runtime benchmarks

### PR DESCRIPTION
**Pull request by Kobzol**
Various simplifications, removing of unused code and fixing clippy issues. Best reviewed commit by commit.

Based on https://github.com/rust-lang/rustc-perf/pull/1606.

### CODE REVIEW

File: collector/src/benchmark/mod.rs [hunk 1]
```diff
@@ -205,7 +205,7 @@ impl Benchmark {
         let iterations = iterations.unwrap_or(self.config.runs);
 
         let profiles: Vec<Profile> = profiles
-            .into_iter()
+            .iter()
             .copied()
```
Comment (nnethercote) [inline collector/src/benchmark/mod.rs:209]: It seems strange to have `into_iter` and `copied` used together. Clippy suggested changing `into_iter` to `iter`, but maybe you could instead keep `into_iter` and remove the `copied`?

File: collector/src/benchmark/mod.rs [hunk 2]
```diff
@@ -216,7 +216,7 @@ impl Benchmark {
         }
 
         let scenarios: Vec<Scenario> = scenarios
-            .into_iter()
```
Comment (nnethercote) [inline collector/src/benchmark/mod.rs:219]: ditto

File: collector/src/benchmark/mod.rs (hunk 1 above)
Comment (nnethercote) [reply collector/src/benchmark/mod.rs:209]: Or maybe this is https://rust-lang.github.io/rust-clippy/master/index.html#/into_iter_on_ref ?

**Review by nnethercote — COMMENTED** [review]
Clippy is getting quite clever. So many unnecessary `&`s!

Should we add Clippy to the CI test workflow, to ensure the code is always Clippy-compliant?

File: site/src/request_handlers/self_profile.rs [hunk 3]
```diff
@@ -582,17 +583,11 @@ pub async fn handle_self_profile(
...
-        )
-        .set(
-            Tag::Metric,
-            selector::Selector::One("cpu-clock".to_string()),
-        );
+    let query = selector::CompileBenchmarkQuery::new()
+        .benchmark(selector::Selector::One(bench_name.to_string()))
+        .profile(selector::Selector::One(profile.parse().unwrap()))
+        .scenario(selector::Selector::One(scenario))
+        .metric(selector::Selector::One(Metric::CpuClock));
```
Comment (nnethercote) [inline site/src/request_handlers/self_profile.rs:590]: nice

**Review by nnethercote — APPROVED** [review]

File: collector/src/benchmark/mod.rs (hunk 1 above)
Comment (nnethercote) [reply collector/src/benchmark/mod.rs:209]: Or maybe this is https://rust-lang.github.io/rust-clippy/master/index.html#/into_iter_on_ref ?

Comment (Kobzol) [reply collector/src/benchmark/mod.rs:209]: Yeah it's `into_iter_on_ref`.

**Pull request merged**